client/v3/lease.go

```diff
@@ -295,25 +295,51 @@ func (l *lessor) KeepAlive(ctx context.Context, id LeaseID) (<-chan *LeaseKeepAl
 	l.mu.Unlock()
 
 	go l.keepAliveCtxCloser(ctx, id, ka.donec)
+
+	errorC := make(chan error, 1)
+	firstKeepAlive := false
+
 	l.firstKeepAliveOnce.Do(func() {
-		go l.recvKeepAliveLoop()
+		firstKeepAlive = true
+		go l.recvKeepAliveLoop(errorC)
 		go l.deadlineLoop()
 	})
 
+	if firstKeepAlive {
```

Comment:
Just move this to body of function passed to `l.firstKeepAliveOnce.Do`

Reply:
Thanks @serathius. It makes sense. Fixed.